synop2bufr/__init__.py

```diff
@@ -116,22 +116,36 @@ def parse_synop(message: str, year: int, month: int) -> dict:
     output['month'] = month
 
     if 'obs_time' in decoded:
-        output['day'] = decoded['obs_time']['day']['value']
-        output['hour'] = decoded['obs_time']['hour']['value']
+        try:
+            output['day'] = decoded['obs_time']['day']['value']
+        except Exception:
+            output['day'] = None
+        try:
+            output['hour'] = decoded['obs_time']['hour']['value']
+        except Exception:
+            output['hour'] = None
 
-    # The minute will be 00 unless specified by exact observation time
+            # The minute will be 00 unless specified by exact observation time
     if 'exact_obs_time' in decoded:
-        output['minute'] = decoded['exact_obs_time']['minute']['value']
+        try:
+            output['minute'] = decoded['exact_obs_time']['minute']['value']
+        except Exception:
+            output['minute'] = None
         # Overwrite the hour, because the actual observation may be from
         # the hour before but has been rounded in the YYGGiw group
-        output['hour'] = decoded['exact_obs_time']['hour']['value']
+        try:
+            output['hour'] = decoded['exact_obs_time']['hour']['value']
+        except Exception:
+            output['hour'] = None
     else:
         output['minute'] = 0
 
     # Translate wind instrument flag from the SYNOP code to the BUFR code
     if 'wind_indicator' in decoded:
-        iw = decoded['wind_indicator']['value']
-
+        try:
```

Comment:
Can replaced with:

```python
iw = decoded['wind_indicator'].get('value')
```

Reply:
We considered using the .get() method, but I believe Dave decided that in a lot of cases it can lead to an ugly expression. So, in favour of being explicit and consistent, try except blocks are always used instead. There shouldn't be a significant difference in performance between both approaches either.